app/controllers/entries_controller.rb

```diff
@@ -4,22 +4,32 @@ class EntriesController < ApplicationController
   DATE_FORMAT = "%d/%m/%Y".freeze
```

Comment:
This would mean 2 controllers with very similar code. I chose not to do this to adhere to the DRY rule. It does make the coding more complicated to read and it contains a few horrible if statements. If you think two seperate controllers is a better solution then i'll do that.